sdk/program/src/lib.rs

```diff
@@ -72,3 +72,107 @@ extern crate serde_derive;
 
 #[macro_use]
 extern crate solana_frozen_abi_macro;
+
+/// Convenience macro for doing integer division where the opersation's safety
+/// can be checked at compile-time
+///
+/// Since `unchecked_div_by_const!()` is supposed to fail at compile-time, abuse
+/// doctests to cover failure modes
+/// Literal denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let _ = unchecked_div_by_const!(10, 0);
+/// # }
+/// ```
+/// #
+/// # Const denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # const D: u64 = 0;
+/// # let _ = unchecked_div_by_const!(10, D);
+/// # }
+/// ```
+/// #
+/// # Non-const denominator fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let d = 0;
+/// # let _ = unchecked_div_by_const!(10, d);
+/// # }
+/// ```
+/// #
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # const N: u64 = 10;
+/// # let _ = unchecked_div_by_const!(N, 0);
+/// # }
+/// ```
+/// #
+/// # Const denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # const N: u64 = 10;
+/// # const D: u64 = 0;
+/// # let _ = unchecked_div_by_const!(N, D);
+/// # }
+/// ```
+/// #
+/// # Non-const denominator fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # const N: u64 = 10;
+/// # let d = 0;
+/// # let _ = unchecked_div_by_const!(N, d);
+/// # }
+/// ```
+/// #
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let n = 10;
+/// # let _ = unchecked_div_by_const!(n, 0);
+/// # }
+/// ```
+/// #
+/// # Const denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let n = 10;
+/// # const D: u64 = 0;
+/// # let _ = unchecked_div_by_const!(n, D);
+/// # }
+/// ```
+/// #
+/// # Non-const denominator fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let n = 10;
+/// # let d = 0;
+/// # let _ = unchecked_div_by_const!(n, d);
+/// # }
+/// ```
+/// #
```

Comment:
Worth having a successful example?

Reply:
Whoops! I deleted the whole `mod tests` block in a fit of defeat.  I'll readd that!